### PR TITLE
 Added "exif" keyword argument to TIFF saving

### LINKED ISSUE #4940: TIFF Image tags are not available
### What did you do?

Tried to access TIFF tags with the Pillow library but get nothing.  
I can upload my .tiff file to https://exifinfo.org/ and see many tags. 

### What did you expect to happen?

To be able to access exif tags of my tiff files.

### What actually happened?

An empty dict is returned.

### What are your OS, Python and Pillow versions?

* OS: Ubuntu 20.04 (WSL)
* Python: 3.7.9
* Pillow: 7.2.0

```python
from PIL import Image

img = Image.open(file_name)
exif_data = img.getexif()
dict(exif_data)
# returns {}
```

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #4940

#5416 allowed `getexif()` to read TIFF `tag_v2` data. This PR allows that data to be saved again, using the `exif` keyword argument.

```python
im.save("out.tiff", exif=im.getexif()
```

### CODE REVIEW

File: src/PIL/TiffImagePlugin.py [hunk 1]
```diff
@@ -1142,6 +1142,17 @@ def load_end(self):
         if not self.is_animated:
             self._close_exclusive_fp_after_loading = True
 
+            # reset buffered io handle in case fp
+            # was passed to libtiff, invalidating the buffer
+            self.fp.tell()
```
Comment (radarhere) [inline src/PIL/TiffImagePlugin.py:1147]: This is copied from https://github.com/python-pillow/Pillow/blob/94ecb279037501e99d1e329634cc1f2cab95cc13/src/PIL/TiffImagePlugin.py#L1075-L1077

**Comment by dbuscombe-usgs** [comment]
Is there some way I can upvote this? There is currently NO library for python (that I know) that will create tiffs from arbitrary arrays and exif data. It would be a huge benefit to a lot of people if PIL would support writing exif in the way that this PR deals with. Thanks!

**Comment by hugovk** [comment]
@dbuscombe-usgs It should be in Pillow 8.4.0, to be released next month.

**Review by hugovk — APPROVED** [review]
Conflict needs resolving.

**Pull request merged**